lib/Cro/HTTP/Client.pm6

```diff
@@ -531,19 +549,25 @@ class Cro::HTTP::Client {
             $http = '';
         }
         my Cro::Uri $proxy-url = self!get-proxy-url($parsed-url);
-        my $request-object = self!assemble-request($method, $parsed-url, $proxy-url, %options);
+        my Cro::Policy::Timeout $timeout-policy;
+        my $request-object = self!assemble-request($method, $parsed-url, $proxy-url, %options, $timeout-policy);
 
         my constant $redirect-codes = set(301, 302, 303, 307, 308);
         my $enable-push = self ?? $!push-promises // %options<push-promises> !! %options<push-promises>;
 
+        my $request-start-time = now;
         Promise(supply {
-            whenever self!get-pipeline($proxy-url // $parsed-url, $http, $request-log, ca => %options<ca>, :$enable-push) -> $pipeline {
+            my $body-timeout = $timeout-policy.get-timeout($request-start-time, 'body');
```

Comment:
This won't work; we can't ask for the body timeout until we reach that phase. Also the parameter to `get-timeout` is the time since the start of the request (and used as such), but here we're passing in an absolute start time.